### ISSUE TITLE
distributed training does not converge as fast as on a single node

### ISSUE THREAD

**Issue by pengjiang030** [SeanNaren/deepspeech.pytorch#282]
Has anyone met the same issue? I thought it was probably because DistributedBucketSampler uses bucket shuffling, so I modified it to a global shuffling. However, the training still stops converging when WER reaches around 22 on AN4. Can anyone share some thoughts on this problem? Thank you!

**Comment by Wilbur529** [comment]
Could you tell me what's the version of python and pytorch you used?

**Comment by ryanleary** [comment]
Are you using a much larger batch size? How many GPUs are you using? If more than 1/changed batch size, have you adjusted learning rate parameters?

**Comment by pengjiang030** [comment]
@Wilbur529 I use python 3.6 and pytorch 0.3.1. Is it related to the versions?

@ryanleary The total minibatch size is 32, the same as training on a single node. So I use 8 GPUs, each computing with a minibatch size of 4. The other configurations are also the same as single node training (learning rate: 3e-4, momentum:0.9, learning-anneal 1.01). It seems the program should behave exactly the same as single node training, but it converges much slower. The only reason I can think of is the bucket shuffling procedure somehow leads to sampling training data with a skewed distribution. What I do not understand is that the 'BucketingSampler' used in single node training also shuffles in the granularity of minibatches; however, it converges much faster and reaches the reported WER after 90 epochs. 

Please correct me if I am wrong.  In the code of BucketingSampler in data/data_loader.py:
```
def __iter__(self):
        for ids in self.bins:
            np.random.shuffle(ids)
            yield ids
```
Is `np.random.shuffle(ids)` necessary? It seems to me that the minibatch indexed by ids will produce the same gradient whether shuffled or not.
